src/middlewared/middlewared/plugins/kerberos.py

```diff
@@ -1227,14 +1235,15 @@ async def check_updated_keytab(self):
 
         old_mtime = 0
         ad_state = await self.middleware.call('activedirectory.get_state')
-        if ad_state == 'DISABLED' or not os.path.exists(keytab['SYSTEM'].value):
+        ad_keytab_exists = await self.middleware.run_in_thread(os.path.exists, keytab['SYSTEM'].value)
+        if ad_state == 'DISABLED' or not ad_keytab_exists:
```

Comment:
This **_used_** to only check the file existence if `ad_state` was != 'DISABLED', so maybe change to 
```
if ad_state == 'DISABLED' or not (await self.middleware.run_in_thread(os.path.exists, keytab['SYSTEM'].value)):
    return
```
... or otherwise avoid adding effort, e..g.
```
if ad_state == 'DISABLED':
    return
ad_keytab_exists = await self.middleware.run_in_thread(os.path.exists, keytab['SYSTEM'].value)
if not ad_keytab_exists:
    return
```